client/repolist.c

```diff
@@ -115,12 +117,27 @@ TDNFLoadRepoData(
 
         TDNF_SAFE_FREE_MEMORY(pszRepoFilePath);
         pszRepoFilePath = NULL;
-
         /* may have added multiple repos, go to last one */
         while (*ppRepoNext)
             ppRepoNext = &((*ppRepoNext)->pNext);
     }
 
+    pRepoParsePre = pReposAll;
+
+    while (pRepoParsePre != NULL && pRepoParsePre->pNext != NULL) {
+        pRepoParseNext = pRepoParsePre;
+ 
+        while (pRepoParseNext->pNext != NULL) {
+            if (!strcmp(pRepoParsePre->pszId, pRepoParseNext->pNext->pszId)) {
+                dwError = ERROR_TDNF_DUPLICATE_REPO_ID;
+                BAIL_ON_TDNF_ERROR(dwError);
+            }
+            else
+                pRepoParseNext = pRepoParseNext->pNext;
+        }
+        pRepoParsePre = pRepoParsePre->pNext;
+    }
```

Comment:
Thank you, this looks good. I liked the version using `for` better though because it makes the algorithm easier to understand (and I can only see it from Shreenidhi's comment).